ql/src/java/org/apache/hadoop/hive/ql/parse/rewrite/MergeRewriter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.parse.rewrite;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.Context;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.Table;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+
+import java.util.Map;
+
+public interface MergeRewriter {
+  void handleSource(boolean hasWhenNotMatchedClause, String sourceAlias, String onClauseAsText);
+
+  void handleWhenNotMatchedInsert(String targetFullName, String columnListText, String hintStr,
+                                  String valuesClause, String predicate, String extraPredicate);
```

Comment:
is that standard formatting?

Reply:
Could you please share the code style of Hive. I saw formatting of many ways in so far

Reply:
i have some style file from downstream :), so it probably doesn't apply here. I usually use 4 spaces(2 tabs) for params that shifted to a next line